README.md

```diff
@@ -20,9 +20,334 @@ Or install it yourself as:
+access_controller.routers
+```
+
+`routers` will return an Array of `Router` objects connected to `access_controller`.
+
+#### Switch Router Auto-Selection
+```ruby
+access_controller.switch_autoselect(flag: 1)
+```
+
+If `flag` is 1, the router auto-selection function will be enabled. If `flag` is 0, the router auto-selection function will be disabled.
+
+#### Open Scanning for all the routers in the router list
+```ruby
+access_controller.open_scan(aps: ["CC:1B:E0:E7:FE:F8",  "CC:1B:E0:E7:FE:F8", "CC:1B:E0:E7:FE:F8"], chip: 0, active: 0, filter_name: "cassia", filter_mac: "CC:1B:E0:E7:FE:F7", filter_uuid: "00001800-0000-1000-800000805f9b34fb")
+```
+
+`aps` is an array of one or multiple router's MAC address.
+`chip` (optional) means which chip to scan.
+`active` (optional) is either 0 or 1. 0 means enable passive scanning, 1 means enable active scanning.
+`filter_name` (optional) is a filter for device name.
+`filter_mac` (optional) is a filter for device MAC.
+`filter_uuid` (optional) is a filter for device UUID.
+
+#### Close Scanning for all the routers in the router list
+```ruby
+access_controller.close_scan(aps: ["CC:1B:E0:E7:FE:F8",  "CC:1B:E0:E7:FE:F8", "CC:1B:E0:E7:FE:F8"])
+```
+ 
+`aps` is an array of one or multiple router's MAC address.
+
+#### Connect A Device to An Automatically Selected Router from the router list
+```ruby
+access_controller.connect_device(aps: '*', device_mac: "CC:1B:E0:E7:FE:F8")
+```
+
+`aps` is an array of one or multiple router's MAC address, or the string '*' which refers to all currently connected routers.
+`device_mac` is the mac address of the device that you are connecting to. You may only pass in one MAC address.
+
+#### Disconnect A Device
+```ruby
+access_controller.disconnect_device(device_mac: "CC:1B:E0:E7:FE:F8")
+```
+
+`device_mac` is the mac address of the device that you are disconnecting.
+
+#### Open Notification on SSE Channel
+```ruby
+access_controller.open_notify(aps:  ["CC:1B:E0:E7:FE:F8","CC:1B:E0:E7:FE:F8"])
+```
+
+`aps` is an array of one or multiple router's MAC address.
+
+#### Close Notification on SSE Channel
+```ruby
+access_controller.close_notify(aps: ["CC:1B:E0:E7:FE:F8","CC:1B:E0:E7:FE:F8"])
+```
+
+`aps` is an array of one or multiple router's MAC address.
+
+#### Open Connection-state Monitoring on SSE Channel
+```ruby
+access_controller.open_connection_state(aps: ["CC:1B:E0:E7:FE:F8","CC:1B:E0:E7:FE:F8"])
+```
+
+`aps` is an array of one or multiple router's MAC address.
+
+#### Close Connection-state Monitoring on SSE Channel
+```ruby
+access_controller.close_connection_state(aps: ["CC:1B:E0:E7:FE:F8","CC:1B:E0:E7:FE:F9"])
+```
+
+`aps` is an array of one or multiple router's MAC address.
+
+#### Open Ap-state Monitoring on SSE Channel
+```ruby
+access_controller.open_ap_state
+```
+
+#### Close Ap-state Monitoring on SSE Channel
+```ruby
+access_controller.close_ap_state
+```
+
+#### Create One Combined SSE Connection
+```ruby
+access_controller.combined_sse
```

Comment:
```
access_controller.combined_sse do |client|
  client.on_event do |event| 
    puts "I received an event: #{event.type}, #{event.data}"
    data = JSON.parse(event.data)
    if data["dataType"] == "scan"
      ac.connect_device(aps: [data["ap"]], device_mac: data["bdaddrs"][0]["bdaddr"])
    end
  end
  client.on_error { |err| puts "#{err.status}" }
end
```

You can pass a block to the `combined_sse` method that allows you to act on the SSE's that you get from the routers. 

We are using the https://github.com/launchdarkly/ruby-eventsource gem `ld-eventsource` for our SSE client. Please refer to their documentation for more info